app/services/userService.js

```diff
@@ -0,0 +1,32 @@
+import uuid from 'uuid/v4'
+import db from '../../database/models'
+import encryptPassword from '../../lib/helpers/encrypt'
+
+class UserService {
+  static async createUser(req, res) {
```

Comment:
```
const userData = addUuid(user) => Object.assign({},
{
  id: uuid(),
  ...userData
})
```
``` 
const hashPassword = userData => 
  Object.assign({ 
    ...userData, 
    password: encryptPassword.generateHash(password) 
  })
```

```
const ensureUserDoesNotExist => findUserFunction => userBody => {
  if(await findUserFunction({ where: { email: userBody.email }}) === null) throw ({ message: 'user already exists' })
  return userBody
}
```
```
const createUser = async userData =>
  await db.User.create(userData)
```

- All this are methods that could be used to clean up the piece of code below, make it more readable and better. Since the way forward is OOP, we could use method chaining to achieve that combined with a bit of functional programming... The endgoal is to leave the codebase better before moving on